roles/provision_virtual_esxi/tasks/main.yml

```diff
@@ -11,25 +11,21 @@
     fail_msg: Variable must be set when using this role.
 
 - name: Check For Missing Hosts
-  community.vmware.vmware_guest_info:
+  vmware.vmware.guest_info:
     hostname: "{{ provision_virtual_esxi_hostname }}"
     username: "{{ provision_virtual_esxi_username }}"
     password: "{{ provision_virtual_esxi_password }}"
     port: "{{ provision_virtual_esxi_port | default(omit) }}"
-    datacenter: "{{ provision_virtual_esxi_datacenter | default('') }}"
     validate_certs: "{{ provision_virtual_esxi_validate_certs | default(omit) }}"
     name: "{{ _esxi.name }}"
-  failed_when: false
   register: _esxi_host_check
   loop: "{{ provision_virtual_esxi_vms }}"
   loop_control:
     loop_var: _esxi
     label: _esxi.name
 
 - name: Include New ESXi Hosts Tasks
-  when: >-
-    (_esxi_host_check.results | selectattr('instance', 'undefined') is any) or
-    (_esxi_host_check.results | selectattr('instance', 'defined') | map(attribute='instance') is not all)
+  when: (_esxi_host_check.results | map(attribute='guests') is not all)
```

Comment:
what about the first part of the condition?

Reply:
the original community module would raise an error if no vm was found matching the search criteria, so we needed the first part of the condition. The guest info module always returns a list, but it will be empty if no VMs were found. So the first check is not needed